salt/utils/http.py

```diff
@@ -247,6 +247,17 @@ def query(
         else:
             http_proxy_url = f"http://{proxy_host}:{proxy_port}"
 
+        if header_dict is None:
+            header_dict = {}
+
+        if method == "POST" and "Content-Type" not in header_dict:
+            log.debug(
+                "Content-Type not provided for POST request, assuming application/x-www-form-urlencoded"
+            )
+            header_dict["Content-Type"] = "application/x-www-form-urlencoded"
+            if "Content-Length" not in header_dict:
+                header_dict["Content-Length"] = f"{len(data)}"
+
```

Comment:
this case covers headers_dict. but what if the user supplies header_list with a content-type, this should go after line 357 where header_list and header_dict were combined into header_dict. that way no setting of content-type if it is going to be set another way. this would also remove he need check if header_dict is none first, as there is already code farther down to do that.

Reply:
Fixed this.